src/main/java/org/truffleruby/core/numeric/FixnumNodes.java

```diff
@@ -900,7 +901,16 @@ public Object leftShiftNeg(long a, int b) {
                 CompilerDirectives.transferToInterpreterAndInvalidate();
                 rightShiftNode = insert(FixnumNodesFactory.RightShiftNodeFactory.create(null));
             }
-            return rightShiftNode.executeRightShift(a, -b);
+            return rightShiftNode.executeRightShift(a, absoluteValue(b));
+        }
+
+        @Specialization(guards = "b < 0")
+        public Object leftShiftNeg(int a, long b) {
+            if (rightShiftNode == null) {
+                CompilerDirectives.transferToInterpreterAndInvalidate();
+                rightShiftNode = insert(FixnumNodesFactory.RightShiftNodeFactory.create(null));
+            }
+            return rightShiftNode.executeRightShift(a, absoluteValue(b));
```

Comment:
To be complete, we should also handle when `b >=0 and is long` as well as combinations where both are long.